packages/plugin-kit/src/lib/decodeSliceMachineConfig.ts

```diff
@@ -0,0 +1,43 @@
+import * as t from "io-ts";
+
+import { SliceMachineConfig } from "../types";
+
+import { decode, DecodeReturnType } from "./decode";
+
+const SliceMachineConfigPluginRegistrationCodec = t.union([
+	t.string,
+	t.intersection([
+		t.type({
+			resolve: t.string,
+		}),
+		t.partial({
+			options: t.UnknownRecord,
+		}),
+	]),
+]);
+
+const SliceMachineConfigCodec = t.intersection([
+	t.type({
+		_latest: t.string,
+		repositoryName: t.string,
+		adapter: SliceMachineConfigPluginRegistrationCodec,
+	}),
+	t.partial({
+		apiEndpoint: t.string,
+		libraries: t.array(t.string),
+		localSliceSimulatorURL: t.string,
+		plugins: t.array(SliceMachineConfigPluginRegistrationCodec),
+	}),
+]);
+
+// TODO: Maybe rename "decode" to "validate". "decode" exposes the `io-ts`
+// internals.
+export const decodeSliceMachineConfig = (
+	input: unknown,
+): DecodeReturnType<SliceMachineConfig, SliceMachineConfig, unknown> => {
+	return decode(SliceMachineConfigCodec, input) as DecodeReturnType<
```

Comment:
It turns out we don't need to use `as`. Thanks for catching that!

It was added when the function was first written because there was some kind of type incompatibility that required asserting the type. Note that we purposely use the `SliceMachineConfig` type rather than the inferred type from `SliceMachineConfigCodec`. This allows us to include TSDoc comments, which is not possible with `io-ts` codecs.